api/src/main/java/org/openmrs/module/labonfhir/api/scheduler/FetchTaskUpdates.java

```diff
@@ -172,19 +181,42 @@ private Boolean updateTasksInBundle(List<Bundle> taskBundles) {
 					// Only update if matching OpenMRS Task found
 					if (openmrsTask != null) {
 						// Handle status
-						openmrsTask.setStatus(openelisTask.getStatus());
+						if (openelisTask.getStatus().toString().equals(TaskStatus.COMPLETED.toString())) {
+
+							openmrsTask.setStatus(openelisTask.getStatus());
 						
-						Boolean taskOutPutUpdated = false;
-						if(openmrsTask.hasBasedOn()){
-                            setOrderNumberFromLIS(openmrsTask.getBasedOn());
+							Boolean taskOutPutUpdated = false;
+							if(openmrsTask.hasBasedOn()){
+								setOrderNumberFromLIS(openmrsTask.getBasedOn());
+							}
+							if (openelisTask.hasOutput()) {
+								// openmrsTask.setOutput(openelisTask.getOutput());
+								taskOutPutUpdated = updateOutput(openelisTask.getOutput(), openmrsTask);
+							}
+							if (taskOutPutUpdated) {
+								taskService.update(openmrsTaskUuid, openmrsTask);
+								tasksUpdated = taskOutPutUpdated;
+							}
+						}
+
+						if(openelisTask.getStatus().toString().equals(TaskStatus.REJECTED.toString()) ){
+							openmrsTask.setStatus(openelisTask.getStatus());
+							commentText  = openelisTask.getStatusReason().getText().toString().isEmpty() ? commentText: openelisTask.getStatusReason().getText();
+                            setOrderStatus(openmrsTask.getBasedOn(), openelisTask.getStatus().toCode(), Order.FulfillerStatus.EXCEPTION, commentText);
+							tasksUpdated = true;
 						}
-						if (openelisTask.hasOutput()) {
-							// openmrsTask.setOutput(openelisTask.getOutput());
-							taskOutPutUpdated = updateOutput(openelisTask.getOutput(), openmrsTask);
+
+						if(openelisTask.getStatus().toString().equals(TaskStatus.CANCELLED.toString()) ){
+							openmrsTask.setStatus(openelisTask.getStatus());
+							commentText  = TaskStatus.CANCELLED.toString();
+                            setOrderStatus(openmrsTask.getBasedOn(), openelisTask.getStatus().toCode(), Order.FulfillerStatus.EXCEPTION, commentText);
+							tasksUpdated = true;
 						}
-						if (taskOutPutUpdated) {
-							taskService.update(openmrsTaskUuid, openmrsTask);
-							tasksUpdated = taskOutPutUpdated;
+						
+						if(openelisTask.getStatus().toString().equals(TaskStatus.REQUESTED.toString()) || openelisTask.getStatus().toString().equals(TaskStatus.ACCEPTED.toString()) ){
+							openmrsTask.setStatus(openelisTask.getStatus());
+                            setOrderStatus(openmrsTask.getBasedOn(), openelisTask.getStatus().toCode(), Order.FulfillerStatus.RECEIVED, commentText);
```

Comment:
this should be `Order.FulfillerStatus.IN_PROGRESS`

Reply:
Done